packages/jest-circus/src/eventHandler.js

```diff
@@ -55,6 +55,23 @@ const eventHandler: EventHandler = (event, state): void => {
         });
       }
 
+      // inherit mode from its parent describe but
+      // do not inherit "only" mode when there is already tests with "only" mode
+      const shouldInheritMode = !(
+        currentDescribeBlock.mode === 'only' &&
+        currentDescribeBlock.tests.find(test => test.mode === 'only')
+      );
+
+      currentDescribeBlock.tests.forEach(test => {
+        if (!test.mode && shouldInheritMode) {
```

Comment:
can wrap the whole statement in `if(shouldInheritMode)` to avoid looping at all if it's falsy